DOCS-6510 Changes (#16)
mulesoft/docs-functional-monitoring: modules/ROOT/pages/afm-monitoring-public-apis.adoc

= Monitoring the Endpoints of Public APIs
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:page-aliases: afm-public-versus-private.adoc
:imagesdir: ../assets/images

> Public APIs are APIs with endpoints that are exposed to the open internet. In API Functional Monitoring (AFM), tests of such endpoints are run by workers that themselves run in public locations or private locations. A _public location_ is a region (which you can think of as a resource pool) that is shared with other MuleSoft customers. Examples of such regions are `us-east-1`, `us-east-2`, and `eu-central-1`. For information about private locations, see xref:afm-monitoring-private-apis.adoc[Monitoring the Endpoints of Private APIs].

Reviewer: ```suggestion
Public APIs are APIs with endpoints that are exposed to the internet. In Anypoint Monitoring API Functional Monitoring (Anypoint AFM), tests of such endpoints are run by workers that themselves run in public locations or private locations. A _public location_ is a region (or resource pool) that is shared with other MuleSoft customers. Examples of such regions are `us-east-1`, `us-east-2`, and `eu-central-1`. For information about private locations, see xref:afm-monitoring-private-apis.adoc[Monitoring the Endpoints of Private APIs].
```

Author: Your suggestion isn't correct.

The tests in monitors are run on schedules. For a worker to run in a public location according to a schedule, a schedule must be associated with it. A public location can be associated with more than one schedule for more than one monitor.

> The following diagram depicts the types of relationship that can exist between monitors, schedules, public locations, APIs, and environments. The different line types depict the different types of relationship.

Reviewer: ```suggestion
The following diagram depicts the types of relationships that can exist between monitors, schedules, public locations, APIs, and environments. The different line types depict the different types of relationships.
```

Reviewer: DITTO edits made in private APIs topic re. bold, colon, and though > although
I did not mark them here

Author: "Though" and "although" are interchangeable as conjunctions, so why substitute the latter for the former?

* *Third relationship type (represented by dashed-and-dotted blue arrows):*
+
Monitor 3 runs on two separate schedules, and each is associated with a different public location. Monitor 3 is written to test one or more endpoints in public API 5 and in public API 6, though it could also test endpoints in more public APIs. The two APIs are in a single environment.

* *Fourth relationship type (represented by dotted orange arrows):*
+
Monitor 4 runs on two separate schedules, and each is associated with the same public location, though each could be associated with a different public location. Each of the two tested public APIs is in a different environment. One environment could be a test environment, and the other environment could be a production environment. Public API 7 and public API 8 are copies of the same API. Monitor 4 is written to test one or more endpoints in public API 7 and the same endpoints in public API 8, though it could also test endpoints in more public APIs.

* *Fifth relationship type (represented by dashed, dark blue lines):*
+

> Monitor 5 runs on a single schedule, though it could run on two. The schedule is associated with Public Location 2. Monitor 5 is written to test one or more endpoints in public API 9 and in public API 10 (though it could also test endpoints in more public APIs), both of which are accessible on the World-Wide Web.

Reviewer: GLOBALLY change World-Wide Web > web

== Length of Time Before Tests Time Out

> To prevent tests run in public locations from consuming too many resources, AFM prevents tests from waiting longer than 120 seconds for a response from an endpoint. You cannot increase this wait time.

Reviewer: ```suggestion
To prevent tests run in public locations from consuming too many resources, API Functional Monitoring prevents tests from waiting longer than 120 seconds for a response from an endpoint. You cannot increase this wait time.
```

== Scheduling Tests

> Tests are run at fixed intervals that you define in schedules. Each test is associated with a schedule. When you run tests against the endpoints of public APIs, AFM sets limitations on the following aspects of scheduling:

Reviewer: ```suggestion
Tests are run at fixed intervals that you define in schedules. Each test is associated with a schedule. When you run tests against the endpoints of public APIs, API Functional Monitoring sets limitations on the following aspects of scheduling:
```

> Number of Schedules::

Reviewer: I wonder at the use of description-list format here, when it is not used for the relationship types.
Seems like the two texts are similar and should use the same format.

+

> By default, AFM lets your organization in Anypoint Platform run up to five schedules for testing from public locations at one time. If you reach the limit of five schedules and want to run more monitors simultaneously, contact your MuleSoft administrator about raising your maximum. You could also run monitors from private locations, if you have access to Anypoint Virtual Private Cloud. There is no limit to the number of schedules that you can create to use private locations. You can monitor public APIs from private locations, too, not just private APIs.

Reviewer: ```suggestion
By default, AFM allows your organization in Anypoint Platform to run up to five simultaneous schedules for testing from public locations. If you want to run more monitors simultaneously, contact your MuleSoft administrator about raising your maximum. 

You can also run monitors from private locations, if you have access to Anypoint Virtual Private Cloud. There is no limit to the number of schedules that you can create to use private locations. 

You can monitor both public and private APIs from private locations.
```

Author: Done.

Duration of Intervals Between Executions of Tests::
+

> The shortest interval at which you can schedule a test to be run is fifteen minutes.

Reviewer: ```suggestion
The shortest interval that you can schedule to occur between tests is fifteen minutes.
```

= Monitoring the Endpoints of Public APIs
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:page-aliases: afm-public-versus-private.adoc
:imagesdir: ../assets/images

> Public APIs are APIs with endpoints that are exposed to the open internet. In API Functional Monitoring (AFM), tests of such endpoints are run by workers that themselves run in public locations. A _public location_ is a region (which you can think of as a resource pool) in CloudHub that is shared with other MuleSoft customers.

Reviewer 2: Public locations are not in Cloudhub. They are actually Lambda Functions. Let's say here that public locations are workers provisioned by Mulesoft, part of the Anypoint Platform and available for everyone to use (shared with others).
We would like not to metion the implementation (Lambda Functions), but specify that we have public locations available in different regions ( **us-east-1**, **us-east-2**, **eu-central-1**)

Reviewer 2: Also you can test public endpoint with private locations.

Author: Resolved.

The tests in monitors are run on schedules. For a worker to run in a public location according to a schedule, a schedule must be associated with it. A public location can be associated with more than one schedule for more than one monitor.

The following diagram depicts the types of relationship that can exist between monitors, schedules, public locations, APIs, and environments. The different line types depict the different types of relationship. 

.The possible relationship types between monitors, schedules, public locations, APIs, and environments

> image::afm-public-locations.png[]

Reviewer 2: Let's change this diagram, as workers are not in clouhud. Let's put them inside AFM, and AFM inside Anypoint platform

Author: Resolved.

Author: Removed the workers from CloudHub. I didn't add Anypoint Platform to the diagram, though.

* *Fourth relationship type (represented by dotted orange arrows):*
+
Monitor 4 runs on two separate schedules, and each is associated with the same public location, though each could be associated with a different public location. Each of the two tested public APIs is in a different environment. One environment could be a test environment, and the other environment could be a production environment. Public API 7 and public API 8 are copies of the same API. Monitor 4 is written to test one or more endpoints in public API 7 and the same endpoints in public API 8, though it could also test endpoints in more public APIs.

* *Fifth relationship type (represented by dashed, dark blue lines):*
+
Monitor 5 runs on a single schedule, though it could run on two. The schedule is associated with Public Location 2. Monitor 5 is written to test one or more endpoints in public API 9 and in public API 10 (though it could also test endpoints in more public APIs), both of which are accessible on the World-Wide Web.

== Length of Time Before Tests Time Out

> To prevent tests run in public locations from consuming too many resources, AFM prevents tests from waiting longer than 120 seconds for a response from an endpoint. You cannot increase this wait time.

Reviewer 2: I would say here that, as this type of locations are shared, we have some limited the max time tests can be executed.

Author: I'm not sure that the info needs to be rephrased as you're suggesting. Would you explain why you'd like me to rephrase it?

== Scheduling Tests

Tests are run at fixed intervals that you define in schedules. Each test is associated with a schedule. When you run tests against the endpoints of public APIs, AFM sets limitations on the following aspects of scheduling:

Number of Schedules::
+

> Initially, AFM lets your organization in Anypoint Platform run up to five workers across public locations at one time. Therefore, you can run up to five tests of the endpoints in public APIs at one time.

Reviewer 2: I would say that: you can only create a limited amount of schedules (5 by default) associated to public locations (as they available for everyone).

Author: Resolved.

+

> The number of workers that are running tests at any particular time is determined by the schedules that you associate with tests. Across the monitors that are created in AFM in your organization, there can be only five schedules. If you reach the limit and want to run more tests simultaneously, contact your MuleSoft administrator about raising your maximum.

Reviewer 2: The number of workers is always the same (this type of worker/location is provisioned by us, thus users cannot create public locations) 
Also I would add that you can create private location if you reach the limit.

Author: Resolved.
